lib/seed/prescription_drug_seeder.rb

```diff
@@ -0,0 +1,80 @@
+module Seed
+  class PrescriptionDrugSeeder
+    def self.call(*args)
+      new(*args).call
+    end
+
+    attr_reader :config
+    attr_reader :counts
+    attr_reader :facility
+    attr_reader :logger
+    attr_reader :user_ids
+    attr_reader :raw_to_clean_medicines
+    delegate :scale_factor, to: :config
+
+    def initialize(config:, facility:, user_ids:)
+      @logger = Rails.logger.child(class: self.class.name)
+      @counts = {}
+      @config = config
+      @facility = facility
+      @user_ids = user_ids
+      @raw_to_clean_medicines = DrugLookup::RawToCleanMedicine.all
+      @logger.info "Starting #{self.class} with #{config.type} configuration"
+    end
+
+    def encounters
+      @facility.encounters
+    end
+
+    def prescription_drugs_to_create
+      (0...5).to_a.sample
+    end
+
+    def call
+      if config.skip_encounters
+        logger.warn { "Skipping seeding prescription drugs, SKIP_ENCOUNTERS is true" }
+        return {}
+      end
+
+      prescription_drugs = []
+      protocol_drugs = if facility.protocol.present?
+        facility.protocol.protocol_drugs
+      else
+        []
+      end
+      encounters.each_with_index do |encounter, index|
```

Comment:
We don't associate drugs to encounters anywhere currently. Shall we create drugs without encounters as the default, and let the user enable encounters for drugs through config if they want?

Reply:
We are not creating encounters for prescription drugs, but creating prescription drugs on the day of the encounter.